### PR TITLE
tikv: document periodic full compaction.

### PR DESCRIPTION
**Pull request by afeinberg**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -267,6 +267,18 @@ This document only describes the parameters that are not included in command-lin
 + Sets the size of the connection pool for service and forwarding requests to the server. Setting it to too small a value affects the request latency and load balancing.
 + Default value: `4`
 
+## raftstore
```
[the remark this answers is not in the document]
Comment (afeinberg) [reply tikv-configuration-file.md]: fixed.

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -1031,6 +1031,16 @@ Configuration items related to Raftstore.
 + Default value: `0.1`
 + Minimum value: `0`
 
+### `periodic-full-compact-start-times` <span class="version-mark">New in 7.6.0</span>
```
[the remark this answers is not in the document]
Comment (Oreoxmt) [reply tikv-configuration-file.md:1034]: @afeinberg PTAL

**Review by Oreoxmt — APPROVED** [review]

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -1031,6 +1031,16 @@ Configuration items related to Raftstore.
...
 
+### `periodic-full-compact-start-times` <span class="version-mark">New in v7.6.0</span>
+
++ Sets the specific times for initiating periodic full compaction. You can specify multiple time schedules in an array. For example, `periodic-full-compact-start-times = ["03:00", "23:00"]` indicates that TiKV performs full compaction at 03:00 AM and 23:00 PM, based on the local time zone of the TiKV node.
++ Default value: `[]`, which means periodic full compaction is disabled by default.
+
+### `periodic-full-compact-start-max-cpu` <span class="version-mark">New in v7.6.0</span>
+
++ Controls the CPU usage threshold at which TiKV performs periodic full compaction.
++ Default value: 0.1, which means that the maximum CPU usage for periodic compaction processes is 10%.
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1042]:
```suggestion
+ Default value: `0.1`, which means that the maximum CPU usage for periodic compaction processes is 10%.
```

File: tikv-configuration-file.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (afeinberg) [reply tikv-configuration-file.md]: @Oreoxmt this one is not dynamically modifiable due via MySQL due to complex data type.

**Comment by afeinberg** [comment]
Verified that we can change the max CPU dynamically. Adding it to dynamic-config.md
```
mysql> show config where type="tikv" and name="raftstore.periodic-full-compact-start-max-cpu";
+------+-----------------+-----------------------------------------------+-------+
| Type | Instance        | Name                                          | Value |
+------+-----------------+-----------------------------------------------+-------+
| tikv | 127.0.0.1:20161 | raftstore.periodic-full-compact-start-max-cpu | 0.5   |
| tikv | 127.0.0.1:20160 | raftstore.periodic-full-compact-start-max-cpu | 0.5   |
| tikv | 127.0.0.1:20162 | raftstore.periodic-full-compact-start-max-cpu | 0.5   |
+------+-----------------+-----------------------------------------------+-------+
3 rows in set (0.01 sec)
```

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -1031,6 +1031,16 @@ Configuration items related to Raftstore.
 + Default value: `0.1`
 + Minimum value: `0`
 
+### `periodic-full-compact-start-times` <span class="version-mark">New in v7.6.0</span>
+
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1035]:
```suggestion

> **Warning:**
>
> Periodic full compaction is experimental. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.

```

File: tikv-configuration-file.md [hunk 5]
```diff
@@ -1031,6 +1031,16 @@ Configuration items related to Raftstore.
...
++ Sets the specific times for initiating periodic full compaction. You can specify multiple time schedules in an array. For example, `periodic-full-compact-start-times = ["03:00", "23:00"]` indicates that TiKV performs full compaction at 03:00 AM and 23:00 PM, based on the local time zone of the TiKV node.
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1036]:
```suggestion
+ Sets the specific times for initiating periodic full compaction. You can specify multiple time schedules in an array. For example, `periodic-full-compact-start-times = ["03:00", "23:00"]` indicates that TiKV performs full compaction at 03:00 AM and 23:00 PM, based on the local time zone of the TiKV node. `periodic-full-compact-start-times = ["03:00 +0000", "23:00 +0000"]` indicates that TiKV performs full compaction at 03:00 AM and 23:00 PM in UTC time.
```

File: tikv-configuration-file.md (hunk 4 above)
Comment (Oreoxmt) [reply tikv-configuration-file.md:1035]: @tonyxuqqi PTAL

File: tikv-configuration-file.md [hunk 6]
```diff
@@ -1031,6 +1031,20 @@ Configuration items related to Raftstore.
 + Default value: `0.1`
 + Minimum value: `0`
 
+### `periodic-full-compact-start-times` <span class="version-mark">New in v7.6.0</span>
+
+> **Warning:**
+>
+> Periodic full compaction is experimental. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
+
++ Sets the specific times for initiating periodic full compaction. You can specify multiple time schedules in an array. For example, `periodic-full-compact-start-times = ["03:00", "23:00"]` indicates that TiKV performs full compaction at 03:00 AM and 23:00 PM, based on the local time zone of the TiKV node. `periodic-full-compact-start-times = ["03:00 +0000", "23:00 +0000"]` indicates that TiKV performs full compaction at 03:00 AM and 23:00 PM in UTC time.
```
Comment (qiancai) [inline tikv-configuration-file.md:1040]:
```suggestion
+ Sets the specific times that TiKV initiates periodic full compaction. You can specify multiple time schedules in an array. For example, `periodic-full-compact-start-times = ["03:00", "23:00"]` indicates that TiKV performs full compaction daily at 03:00 AM and 23:00 PM, based on the local time zone of the TiKV node. `periodic-full-compact-start-times = ["03:00 +0000", "23:00 +0000"]` indicates that TiKV performs full compaction daily at 03:00 AM and 23:00 PM in UTC time.
```

File: tikv-configuration-file.md [hunk 7]
```diff
@@ -1031,6 +1031,20 @@ Configuration items related to Raftstore.
...
++ Default value: `[]`, which means periodic full compaction is disabled by default.
+
+### `periodic-full-compact-start-max-cpu` <span class="version-mark">New in v7.6.0</span>
+
++ Controls the CPU usage threshold at which TiKV performs periodic full compaction.
```
Comment (qiancai) [inline tikv-configuration-file.md:1045]:
```suggestion
+ Limits the maximum CPU usage rate for TiKV periodic full compaction.
```

**Comment by afeinberg** [comment]
@qiancai @Oreoxmt thanks for the last updates. This looks good, do I need to do anything else for this to be merged?

File: tikv-configuration-file.md [hunk 8]
```diff
@@ -1031,6 +1031,20 @@ Configuration items related to Raftstore.
 + Default value: `0.1`
 + Minimum value: `0`
 
+### `periodic-full-compact-start-times` <span class="version-mark">New in v7.6.0</span>
+
+> **Warning:**
+>
+> Periodic full compaction is experimental. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
+
++ Sets the specific times that TiKV initiates periodic full compaction. You can specify multiple time schedules in an array. For example, `periodic-full-compact-start-times = ["03:00", "23:00"]` indicates that TiKV performs full compaction daily at 03:00 AM and 23:00 PM, based on the local time zone of the TiKV node. `periodic-full-compact-start-times = ["03:00 +0000", "23:00 +0000"]` indicates that TiKV performs full compaction daily at 03:00 AM and 23:00 PM in UTC time.
```
Comment (qiancai) [inline tikv-configuration-file.md:1040]:
```suggestion
+ Sets the specific times that TiKV initiates periodic full compaction. You can specify multiple time schedules in an array. For example, `periodic-full-compact-start-times = ["03:00", "23:00"]` indicates that TiKV performs full compaction daily at 03:00 AM and 11:00 PM, based on the local time zone of the TiKV node. `periodic-full-compact-start-times = ["03:00 +0000", "23:00 +0000"]` indicates that TiKV performs full compaction daily at 03:00 AM and 11:00 PM in UTC time.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Comment by Oreoxmt** [comment]
@afeinberg Thanks for your contribution. I will merge this PR after code freeze (Jan 8, 2024)

**Comment by afeinberg** [comment]
> @afeinberg Thanks for your contribution. I will merge this PR after code freeze (Jan 8, 2024)

Thanks, @Oreoxmt

**Comment by afeinberg** [comment]
@Oreoxmt it is now 1/9/2024. Can you get this merged? Let me know if there's anything else I can do.

**Pull request merged**